lib/beautiful/xresources.lua

```diff
@@ -119,13 +130,23 @@ function xresources.set_dpi(dpi, s)
     end
 end
 
+--- Set a rounding mode for autodetected DPIs
+--
+-- If set, beautiful will round the screen DPI to a multiple of the reference
+-- 96 DPI, according to the function used
+-- @tparam function func a function to be applied to the DPI scaling factor
+function xresources.set_dpi_rounding(func)
+    dpi_scale_rounding = func
```

Comment:
Sorry, but this function is useless. It gets as argument the value `dpi/96`. If you want to e.g. ceil this, you could just do `set_dpi(s, 96*math.ceil(get_dpi(s)/96))` in your rc.lua (or just compute the wanted DPI value by hand). So, this does not add any new possibilities.

Put differently: Isn't it enough to be able to override DPI per-screen?

If you want a rounding function, I would more expect the user to override the `round` call in `apply_dpi`. However, that function must return integers, so this might not be all that useful either.